src/main/java/be/seeseemelk/mockbukkit/ServerMock.java

```diff
@@ -236,6 +236,17 @@ public void registerEntity(@NotNull EntityMock entity)
 		entities.add(entity);
 	}
 
+	/**
+	 * Unregisters an entity from the server. Should only be used internally.
+	 *
+	 * @param entity The entity to unregister
+	 */
+	public void unregisterEntity(@NotNull EntityMock entity)
+	{
+		AsyncCatcher.catchOp("entity remove");
+		entities.remove(entity);
```

Comment:
Perhaps this should check if the entity is valid, and remove it if it is? Despite it being labeled for internal use only, people will still try to use it and I'm sure ask for help when it causes weird issues.

Reply:
Was this intended to be marked as resolved?

Reply:
A precondition has been added to check if the entity is not valid, in another word if the `remove` method was called first. The server should continue to track any valid entities, right?

Reply:
Ah, I somehow looked over that; thought just the null check was added. Yeah, that's fine.